tidy3d/components/source.py

```diff
@@ -256,14 +256,28 @@ class GaussianPulse(Pulse):
     >>> pulse = GaussianPulse(freq0=200e12, fwidth=20e12)
     """
 
+    allow_dc_component: bool = pydantic.Field(
+        False,
+        title="Allow DC Component",
+        description="Whether to allow a DC component in the Gaussian pulse spectrum. "
+        "If ``False``, the Gaussian pulse is modified to zero out the DC component, which is "
+        "usually desirable so that the fields will decay. However, for broadband simulations, "
+        "it may be better to have non-vanishing source power near zero frequency. Setting "
+        "this to ``True`` results in an unmodified Gaussian pulse spectrum which can have "
+        "a nonzero DC component.",
+    )
+
     def amp_time(self, time: float) -> complex:
         """Complex-valued source amplitude as a function of time."""
 
         twidth = 1.0 / (2 * np.pi * self.fwidth)
         omega0 = 2 * np.pi * self.freq0
         time_shifted = time - self.offset * twidth
 
-        const = 1j + time_shifted / twidth**2 / omega0
+        if self.allow_dc_component:
+            const = 1
+        else:
+            const = 1j + time_shifted / twidth**2 / omega0
```

Comment:
this is just personal preference so feel free to ignore, but I feel it makes it slightly better to add the DC correction as its own block of code kind of like this.

```py
pulse_amp = offset * oscillation * amp

# subtract out DC component
if not self.allow_dc_component:
    pulse_amp *= 1j + time_shifted / twidth**2 / omega0
return pulse_amp
```